src/core/ext/transport/chaotic_good/data_endpoints.cc

```diff
@@ -246,25 +247,21 @@ auto Endpoint::ReadLoop(uint32_t id, RefCountedPtr<InputQueue> input_queues,
 Endpoint::Endpoint(uint32_t id, RefCountedPtr<OutputBuffers> output_buffers,
                    RefCountedPtr<InputQueue> input_queues,
                    PendingConnection pending_connection, bool enable_tracing,
-                   grpc_event_engine::experimental::EventEngine* event_engine,
-                   std::shared_ptr<GlobalStatsPluginRegistry::StatsPluginGroup>
-                       stats_plugin_group) {
+                   TransportContextPtr ctx) {
   auto arena = SimpleArenaAllocator(0)->MakeArena();
-  arena->SetContext(event_engine);
+  arena->SetContext(ctx->event_engine.get());
   party_ = Party::Make(arena);
   party_->Spawn(
       "write",
       [id, enable_tracing, output_buffers = std::move(output_buffers),
        input_queues = std::move(input_queues),
        pending_connection = std::move(pending_connection),
-       arena = std::move(arena),
-       stats_plugin_group = std::move(stats_plugin_group)]() mutable {
+       arena = std::move(arena), ctx = std::move(ctx)]() mutable {
         return TrySeq(
             pending_connection.Await(),
             [id, enable_tracing, output_buffers = std::move(output_buffers),
```

Comment:
nit: This labda has so many parameters, including another inline lambda declaration... It is hard to follow, consider extracting to a class.